pkg/util/git.go

```diff
@@ -215,61 +198,51 @@ func parseBitbucketUrl(g GitUrl, url *url.URL, c http.Client) (GitUrl, error) {
 		}
 	}
 
-	if !isGitUrlPublic(g, c) {
-		g.token = os.Getenv(BitbucketToken)
-	}
-
-	return g, err
+	return err
 }
 
-func isGitUrlPublic(g GitUrl, c http.Client) bool {
-	host := g.Host
-	if host == RawGitHubHost {
-		host = GitHubHost
-	}
-
-	repo := fmt.Sprintf("%s://%s/%s/%s", g.Protocol, host, g.Owner, g.Repo)
-
-	if res, err := c.Get(repo); err != nil {
-		return false
-	} else if res.StatusCode == http.StatusOK {
-		return true
-	}
-	return false
-}
-
-// CloneGitRepo clones a GitHub Repo to a destination directory
+// CloneGitRepo clones a git repo to a destination directory
 func CloneGitRepo(g GitUrl, destDir string) error {
-	var cloneOptions *gitpkg.CloneOptions
-
 	host := g.Host
 	if host == RawGitHubHost {
 		host = GitHubHost
 	}
 
+	isPublic := true
 	repoUrl := fmt.Sprintf("%s://%s/%s/%s.git", g.Protocol, host, g.Owner, g.Repo)
-	branch := fmt.Sprintf("refs/heads/%s", g.Branch)
 
-	cloneOptions = &gitpkg.CloneOptions{
-		URL:           repoUrl,
-		ReferenceName: plumbing.ReferenceName(branch),
-		SingleBranch:  true,
-		Depth:         1,
+	params := HTTPRequestParams{
+		URL: repoUrl,
 	}
 
-	if g.token != "" {
-		cloneOptions.Auth = &githttp.BasicAuth{
-			// go-git auth allows username to be anything except
-			// an empty string for GitHub and GitLab, however requires
-			// for Bitbucket to be "x-token-auth"
-			Username: "x-token-auth",
-			Password: g.token,
+	// check if the git repo is public
+	_, err := HTTPGetRequest(params, 0)
+	if err != nil {
+		// private git repo requires authentication
+		isPublic = false
```

Comment:
We can't rely on an error to say the the repo is private.  We need to make a GET call with auth to determine that